torch/csrc/distributed/rpc/python_functions.cpp

```diff
@@ -162,45 +161,46 @@ PyRRef pyRemoteBuiltin(
   auto scriptRemoteCall = std::make_unique<ScriptRemoteCall>(
       op, std::move(stack), userRRef->rrefId(), userRRef->forkId());
 
+  auto agent = RpcAgent::getDefaultRpcAgent();
   auto fm = sendMessageWithAutograd(
-      agent, dst, std::move(*scriptRemoteCall).toMessage(), false, rf);
+      *agent, dst, std::move(*scriptRemoteCall).toMessage(), false, rf);
 
   ctx.addPendingUser(userRRef->forkId(), userRRef);
   fm->addCallback(callback::confirmPendingUser);
   return PyRRef(userRRef);
 }
 
 std::shared_ptr<FutureMessage> pyRpcPythonUdf(
-    RpcAgent& agent,
     const WorkerInfo& dst,
     std::string& pickledPythonUDF,
     std::vector<torch::Tensor>& tensors,
     const std::shared_ptr<torch::autograd::profiler::RecordFunction>& rf) {
   auto pythonCall = std::make_unique<PythonCall>(
       std::vector<char>(pickledPythonUDF.begin(), pickledPythonUDF.end()),
       tensors);
+
+  auto agent = RpcAgent::getDefaultRpcAgent();
   return sendMessageWithAutograd(
-      agent,
+      *agent,
       dst,
       std::move(*pythonCall).toMessage(),
       true /*forceGradRecording*/,
       rf);
 }
 
 PyRRef pyRemotePythonUdf(
-    RpcAgent& agent,
     const WorkerInfo& dst,
     std::string& pickledPythonUDF,
     std::vector<torch::Tensor>& tensors,
     const std::shared_ptr<torch::autograd::profiler::RecordFunction>& rf) {
   auto& ctx = RRefContext::getInstance();
   auto serializedPyObj =
       SerializedPyObj(std::move(pickledPythonUDF), std::move(tensors));
+  auto agent = RpcAgent::getDefaultRpcAgent();
```

Comment:
is this agent used anywhere below in the new code?

Reply:
no. should remove it after changing sendPythonRemoteCall.